wallets/index.js

```diff
@@ -251,9 +255,9 @@ function useConfig (wallet) {
         valid = false
       }
 
-      if (valid) await setServerConfig(newServerConfig, { priorityOnly })
+      if (valid) await setServerConfig(newServerConfig, { priorityOnly: skipTests })
     }
-  }, [hasClientConfig, hasServerConfig, setClientConfig, setServerConfig, wallet])
+  }, [config, hasClientConfig, hasServerConfig, setClientConfig, setServerConfig, wallet])
```

Comment:
having `config` as dependency in saveConfig causes the callback to be triggered twice.
Not sure why, could be a race condition because `saveConfig` side effect is updating `config`, or could be because some code upstream is depending on the `saveConfig` state.

Removing config as dependency solves the priority issue

Reply:
Ah, cool, thanks, that would make sense. But removing `config` as a dependency would break this:

```js
if (typeof newConfig === 'function') {
  newConfig = newConfig(config)
}
```

which I added to no longer access local storage directly in `useWallet` and use `saveConfig` instead but I think I could have done simply this:

```diff
   const enablePayments = useCallback(() => {
-    saveConfig(config => ({ ...config, enabled: true }), { skipTests: true })
+    saveConfig(({ ...config, enabled: true }), { skipTests: true })
     logger.ok('payments enabled')
     disableFreebies().catch(console.error)
-  }, [saveConfig, logger])
+  }, [config, logger])

   const disablePayments = useCallback(() => {
-    saveConfig(config => ({ ...config, enabled: false }), { skipTests: true })
+    saveConfig(({ ...config, enabled: false }), { skipTests: true })
     logger.info('payments disabled')
-  }, [name, me, logger])
+  }, [config, name, me, logger])
```

I will test this out later.